packages/xo-server/src/xapi/mixins/patching.mjs

```diff
@@ -629,7 +629,7 @@ export default {
         continue
       }
 
-      const residentVms = host.$resident_VMs.map(vm => vm.uuid)
```

Comment:
What was the issue?

Reply:
I added the zammad ticket in the description.
`host.$resident_VMs` return `resident_VMs` before the `host.evacuate`, So `if(residentVms.includes(vmId))` is always true, and don't migrate VMs.

Reply:
I added the zammad ticket in the description.
`host.$resident_VMs` return `resident_VMs` before the `host.evacuate`, So `if(residentVms.includes(vmId))` is always true, and don't migrate VMs.

Reply:
Do you mean that `host.$resident_VMs` is out of date?

In that case, shouldn't we simply add a `barrier()`?

Reply:
Yes.